internal/blueprint/customizations_test.go

```diff
@@ -0,0 +1,158 @@
+package blueprint
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestCustomizationsHostname(t *testing.T) {
+
+	var myHostname = "MyHostname"
+
+	TestCustomizations := Customizations{
+		Hostname: &myHostname,
+	}
+
+	bpOrig := Blueprint{
+		Name:           "customizations-test",
+		Description:    "Testing customizations functions",
+		Version:        "0.0.1",
+		Customizations: &TestCustomizations,
+	}
+
+	assert.Equalf(t, &myHostname, TestCustomizations.GetHostname(), "The hostname is different")
```

Comment:
+1 here, one of the benefits we get from the assert package is that it tells you what was expected (the 1st argument) and what was the actual value (2nd argument) in case of failure. The error message doesn't really bring out any additional information. 

You can experiment with this by changing the expected hostname after calling GetHostname & before asserting the result to see how it looks like.

Reply:
@ondrejbudai it is still not clear to me what the difference between .Equal and .Equalf is. It looks like both accept a format string for error messages. Maybe we need to document this in test/README once we figure it out.

Reply:
Yes that should be enough, thanks.

Reply:
> @ondrejbudai it is still not clear to me what the difference between .Equal and .Equalf is. It looks like both accept a format string for error messages. Maybe we need to document this in test/README once we figure it out.

I have found one difference. `.Equal` **can** accepted a formatted string but `.Equalf` **requires** it. I don't know what the use case for this would be though.

Reply:
If you use a format string, you should use `.Equalf`, the reason is that some linters will compalain if you use one in `.Equal`, and that is why `.Equalf` was added (and I suppose the old behaviour was kept for compatibility).